peer/peer.go

```diff
@@ -227,6 +227,12 @@ type Config struct {
 	// form "major.minor.revision" e.g. "2.6.41".
 	UserAgentVersion string
 
+	// UserAgentComments specifies any additional comments to include the in
+	// advertised user agent to advertise.  This is optional, so it may be
+	// nil.  If specified, the comments should only exists of characters
+	// from the semantic alphabet [a-zA-Z0-9-].
```

Comment:
Is this comment about the semver alphabet correct for bitcoin-style useragents?

Reply:
It's not actually well-defined since there was no BIP or anything associated with it.  However, in practice, since the comments are delimited by semicolons, everyone always sticks to alphanumerics and the dash, which is the semantic alphabet.

I added that comment because, for Decred, I think we should make it a point to better define its semantics.